src/keyring/KeyringController.ts

```diff
@@ -369,11 +368,11 @@ export class KeyringController extends BaseController<KeyringConfig, KeyringStat
   /**
    * Signs a transaction by calling down into a specific keyring
    *
-   * @param transaction - Transaction object to sign
+   * @param transaction - Transaction object to sign. Must be a `ethereumjs-tx` transaction instance.
```

Comment:
We could create a type for this representing the minimum set of fields required

https://www.typescriptlang.org/play?ssl=23&ssc=2&pln=1&pc=1#code/C4TwDgpgBAsglgOzgWwK7ICoA8DyAjAKwgGNgoBeKAbwCgopgsAFAQwCcXkBnALmrvpQAJi2As+XYG0QBzADQD6wAPYSpshYKgAzNsuRrpCeYqgyWvKJKMmt5rk2nEIhjaYBuLADaoXV9caaUAC+QQDaysAAFhBsANIQIJbWsgC6fKgIANYIygDuCArBNDTEygiSUFh88EhomLiEJGSUtErM7JyWbYIiYnwA5AAMWDFeXvnKbF5CA0FKqlDDWCxCQmwQXFwAjHOmuvqDI6vrm1wATHt2FkdYQ1eC9o5wzrf381CePn7L7wKhAhYADlygAlCAAR1QcA2QgSIEG2yGf2CQA

I'm ambivalent on whether we do this here or in a future change.

Reply:
Good idea. I'm somewhat inclined to leave it for now though, as fixing up the keyring controller is on the path to snaps anyway. I'm anticipating that we'll be changing this method signature to be more type-friendly in the near future, and removing the dependence upon `ethereumjs-tx`. At least I want to, now that I've discovered this!